src/Compilers/CSharp/Portable/Emitter/Model/SynthesizedPrivateImplementationDetailsStaticConstructor.cs

```diff
@@ -21,7 +24,34 @@ internal override void GenerateMethodBody(TypeCompilationState compilationState,
         {
             SyntheticBoundNodeFactory factory = new SyntheticBoundNodeFactory(this, this.GetNonNullSyntaxNode(), compilationState, diagnostics);
             factory.CurrentMethod = this;
-            
+
+            ArrayBuilder<BoundStatement> body = ArrayBuilder<BoundStatement>.GetInstance();
```

Comment:
It looks like it should be possible to accurately estimate amount of items we will need to add to the builder.